tests/test_xarray.py

```diff
@@ -485,3 +485,32 @@ def test_mixed_grouping(chunk):
         fill_value=0,
     )
     assert (r.sel(v1=[3, 4, 5]) == 0).all().data
+
+
+@pytest.mark.parametrize("dtype_out", [np.float64, "float64", np.dtype("float64")])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("chunk", (True, False))
+def test_dtype(chunk, dtype, dtype_out, engine):
+    if chunk and not has_dask:
+        pytest.skip()
+
+    if chunk:
+        data = dask.array.ones((4, 12), dtype=dtype, chunks=(1, -1))
+    else:
+        data = np.ones((4, 12), dtype=dtype)
+
+    arr = xr.DataArray(
+        data,
+        dims=("x", "t"),
+        coords={
+            "labels": ("t", np.array(["a", "a", "c", "c", "c", "b", "b", "c", "c", "b", "b", "f"]))
+        },
+        name="arr",
+    )
+    actual = xarray_reduce(arr, "labels", func="mean", dtype=dtype_out)
+    assert actual.dtype == np.dtype("float64")
+    assert actual.compute().dtype == np.dtype("float64")
```

Comment:
You could test with `xr.set_options(use_flox=False)` as well and compare that the values also are the same.